tests/Benchmarks/System.Text.JsonLab/JsonReaderPerf.cs

```diff
@@ -2,70 +2,277 @@
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
 using BenchmarkDotNet.Attributes;
+using BenchmarkDotNet.Attributes.Jobs;
 using Benchmarks;
 using System.Buffers.Text;
 using System.Collections.Generic;
 using System.IO;
 
-using static System.Text.JsonLab.Benchmarks.Helper;
-
 namespace System.Text.JsonLab.Benchmarks
 {
+    public enum TestCaseType
+    {
+        HelloWorld,
+        Basic,
+        BasicLargeNum,
+        SpecialNumForm,
+        ProjectLockJson,
+        FullSchema1,
+        FullSchema2,
+        DeepTree,
+        BroadTree,
+        LotsOfNumbers,
+        LotsOfStrings,
+        Json400B,
+        Json4KB,
+        Json40KB,
+        Json400KB
+    }
+
+    // Since there are 120 tests here (4 * 2 * 15), setting low values for the warmupCount, targetCount, and invocationCount
+    [SimpleJob(-1, 3, 5, 1024)]
     [MemoryDiagnoser]
     public class JsonReaderPerf
     {
-        private byte[] _data;
-        private SymbolTable _symbolTable;
+        private string _jsonString;
+        private byte[] _dataUtf8;
+        private byte[] _dataUtf16;
         private MemoryStream _stream;
         private StreamReader _reader;
 
-        [Params(EncoderTarget.InvariantUtf8, EncoderTarget.InvariantUtf16)]
-        public EncoderTarget Target;
+        [Params(true, false)]
+        public bool IsUTF8Encoded;
 
-        [ParamsSource(nameof(ValuesForJsonString))]
-        public string JsonString;
+        [ParamsSource(nameof(TestCaseValues))]
+        public TestCaseType TestCase;
 
-        public static IEnumerable<string> ValuesForJsonString() => new[] { JsonStrings.HeavyNestedJson, JsonStrings.HelloWorld };
+        public static IEnumerable<TestCaseType> TestCaseValues() => (IEnumerable<TestCaseType>)Enum.GetValues(typeof(TestCaseType));
 
         [GlobalSetup]
         public void Setup()
         {
-            _symbolTable = GetTargetEncoder(Target);
-            _data = EncodeTestData(Target, JsonString);
+            switch (TestCase)
```

Comment:
this switch grew really big. I would either make it a dictionary to save some space or name enums and string resource keys in the same way to have sth like `_jsonString = JsonStrings.GetString(TestCase.ToString());`